nixos/modules/services/audio/mpd.nix

```diff
@@ -123,10 +133,24 @@ in {
 
   config = mkIf cfg.enable {
 
+    systemd.sockets.mpd = mkIf cfg.startWhenNeeded {
+      description = "Music Player Daemon Socket";
+      wantedBy = [ "sockets.target" ];
+      listenStreams = [
+        "/run/mpd/socket"
+        "${cfg.network.listenAddress}:${toString cfg.network.port}"
+      ];
```

Comment:
I have now and the upstream MPD socket file does the same. I decided to use `/var/run/mpd/socket` to use the same location MPD uses.

While testing I noticed that `any` for `listenAddress` needed to be handled with care which is fixed now.